client/i18n/README.md

```diff
@@ -0,0 +1,120 @@
+
+```tsx
+import {useTranslation} from 'react-i18next';
+...
+export const myComponent = () => {
+    const {t: translate} = useTranslation();
+    return (
+        <>
+            <div>{translate('my.key')}</div>
+            <div>{translate('with.inserts', {valueToInsert: "test"})}</div>
+        </>);
+}
+```
+
+### Providing translation values
+
+`Bootstrap/i18n.ts` configures the `i18next` library to load translations from `./index.ts`.
+
+`index.ts` exports an Object containing translations from `<language>.json`.
+
+`<language>.json` contains the following shape - where `{{key}}` is substituted using a matching key in the inserts passed to `translate()`:
+
+```json
+{
+  "my": {
+    "key": "value"
+  },
+  "with": {
+    "inserts": "This has a {{valueToInsert}}"
+  }
+}
+```
+
+#### Fallback
+
+If you're using the `t` function - the key will be rendered if the translation is missing in the desired language or any fallbacks. Inserts are ignored.
+
+If you're using the `Trans` component, any children will be rendered but not the `ii18nkey` prop.
+
+## Basic formatting in translations
+
+The `<Trans>` component can be used to render translated messages that contain basic formatting tags - `<br>, <strong>, <i>, <p>` can all be embedded directly into the translation string. `<Trans>` takes `translate` as a property `t` to allow it to re-render when language changes. `i18nKey` is passed in to point to the right translation. Inserts are passed as singleton key:value objects.
+
+```tsx
+import {Trans, useTranslation} from 'react-i18next';
+export const myComponent = () => {
+    const {translate} = useTranslation();
+    return (
+        <>
+            <div><Trans t={translate} i18nKey="simple.format></Trans></div>
+            <div><Trans t={translate} i18nKey="with.inserts>{{insert: "value1"}}{{another:"value2"}}</Trans></div>
+
+        </>);
+}
+```
+
+### Providing translation values
+
+```json
+{
+  "simple": {
+    "format": "This is a <strong>value</strong>"
+  },
+  "with": {
+    "inserts": "This has an <i>{{insert}}</i> and {{another}}"
+  }
+}
+```
+
+## Advanced formatting
+
+The `<Trans>` component also supports interpolation of JSX elements into the translated string to provide additional formatting (e.g class names, headers, links) around the string content. These elements must be passed as children to `<Trans>`. Dynamic content should be supplied after elements as singleton children as before.
+
+```tsx
+import {Trans, useTranslation} from 'react-i18next';
+export const myComponent = () => {
+    const {t: translate} = useTranslation();
+    return (
+        <>
+            <div><Trans t={translate} i18nKey="with.link><a href="."/></Trans></div>
+            <div><Trans t={translate} i18nKey="with.classname><div className="bold"/>{{topic: "mytopic"}}</Trans></div>
+            <div><Trans t={translate} i18nKey="with.custom><MyComp prop="value"/><SecondComp/></Trans></div>
```

Comment:
It would be nice to have a more complete example of passing in complex components like `MyComp` - but this can be a subsequent issue I think.

Reply:
Ah, actually this is covered in the next section.